src/nexus/utils.py

```diff
@@ -38,9 +42,10 @@ def get_total_records(self, endpoint: str) -> int:
                 "Range": "0-0",
                 "Prefer": "count=exact",
             },
+            params=params,
             timeout=self.request_timeout_seconds
         )
-        if response.status_code == requests.codes.partial_content:
+        if response.status_code == requests.codes.partial_content or response.status_code == requests.codes.ok:
```

Comment:
Looking good now. I would even suggest a shorter alternative (since this is in a single line):

```suggestion
        if response.status_code in (requests.codes.partial_content, requests.codes.ok):
```

Reply:
updated @ricardogsilva